### ISSUE TITLE
Project with service provider of type "service" and cli asking for passphrase after updating > 3.1.0 

### ISSUE THREAD

**Issue by andreujuanc** [pulumi/pulumi#9117]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details
I've got a project in which we migrated from v2.x to v3 today. It was all good except that after moving to v3 we noticed that the cli is requiring a passphrase which we don't use.
This is the section of the state that configures our secrets provider:
```json
    "secrets_providers": {
            "type": "service",
            "state": {
                "url": "https://api.pulumi.com",
                "owner": "OWNERNAME",
                "project": "profile-service",
                "stack": "dev"
            }
        },
```
I've set the `PULUMI_CONFIG_PASSPHRASE` to `test` and it ran. Then set it to `test2` and also deployed it all.

Then i rebuild the container (I'm running within a docker container).
And It still asks me for the passphrase. Even if I type anything I get this:
![image](https://user-images.githubusercontent.com/360052/156825136-16c40811-f0cc-4dcd-b91e-dc8768824327.png)

So I have to set the PULUMI_CONFIG_PASSPHRASE to any value for the deploys to work.
Setting the env var and deploying has no effect on the `secrets_provider` section of my state.

Just to clarify, I never configured my project with passphrase, I used the default (pulumi backend).

Expected: 
Using a provider other than passphrase should not ask for it.

Actual: 
We get a prompt asking for a passphrase and|or the env vars.

- Creating a new stack withing the same project gets me the same issue.

- But I cannot reproduce it from a brand new project.  I have compared states and versions and they are identical. 

Any help would be appreciated.

**Comment by andreujuanc** [comment]
UPDATE: `pulumi destroy` runs without issues. It does not require the PULUMI_CONFIG_PASSPHRASE

**Comment by andreujuanc** [comment]
The actual problem is that I read the state of another project which in 1 environment uses passphrase.
Just running `new pulumi.StackReference(`<Project Path that uses passphrase>`)` will trigger the passphrase prompt. 
The weird part is that this was not the case before, but only after v3. And also not even reading secrets (which I get should be protected) but these are "open". 
When I go to the pulumi dashboard I don't have to enter the passphrase to read the non protected values.

**Comment by lunaris** [comment]
I think this is a duplicate of/related to and will be closed alongside #11109
